optaplanner-examples/src/main/java/org/optaplanner/examples/conferencescheduling/persistence/ConferenceSchedulingImporter.java

```diff
@@ -0,0 +1,496 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.examples.conferencescheduling.persistence;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.StringReader;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import javax.json.Json;
+import javax.json.JsonArray;
+import javax.json.JsonObject;
+import javax.json.JsonReader;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.optaplanner.examples.conferencescheduling.domain.ConferenceParametrization;
+import org.optaplanner.examples.conferencescheduling.domain.ConferenceSolution;
+import org.optaplanner.examples.conferencescheduling.domain.Room;
+import org.optaplanner.examples.conferencescheduling.domain.Speaker;
+import org.optaplanner.examples.conferencescheduling.domain.Talk;
+import org.optaplanner.examples.conferencescheduling.domain.TalkType;
+import org.optaplanner.examples.conferencescheduling.domain.Timeslot;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ConferenceSchedulingImporter {
+
+    private static final Logger logger = LoggerFactory.getLogger(ConferenceSchedulingImporter.class);
+    private static final String zoneId = "Europe/Paris";
+    private static final String[] smallRoomsTypeNames = {"Quickie Sessions", "Quickie", "Hands-on Labs", "BOF (Bird of a Feather)"};
```

Comment:
static finals are constants, constanstans need to use capitals, so SMALL_ROOMS_TYPE_NAMES instead

Reply:
Just for the record, not all static final variables are constants. For example you can have a static final collection, which, unless you create it as unmodifiable, can be easily changed (added and removed to) and therefore is not a constant and shouldn't use the all capitals convention.

The deciding factor of course is the author's intention about the purpose of the variable.

Another example is the logger field, which is always kept lower case in optaplanner code, although it is a static final variable. I think that is correct, since, no matter if the logger has an internal state that's being modified or not, it is not intended to be used as a constant.